coriolis/tests/api/middleware/test_auth.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2022 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+from unittest import mock
+
+from oslo_middleware import request_id
+from oslo_serialization import jsonutils
+
+from coriolis.api.middleware.auth import CoriolisKeystoneContext
+from coriolis.api import wsgi
+from coriolis import context
+from coriolis.tests import test_base
+
+
+class CoriolisKeystoneContextTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis api middleware auth."""
+
+    @mock.patch.object(jsonutils, "loads")
+    @mock.patch.object(context, "RequestContext")
+    def test__call__(self, mock_request_context, mock_jsonutils_loads):
+        req_mock = mock.Mock()
+        headers_mock = mock.Mock()
+        environ_mock = {}
+        remote_addr_mock = mock.Mock()
+
+        req_mock.headers = headers_mock
+        req_mock.environ = environ_mock
+        req_mock.remote_addr = remote_addr_mock
+
+        mock_user = 'mock_user'
+        mock_id = 'mock_id'
+        mock_project_id = '1'
+        mock_project_name = 'mock_project_name'
+        mock_project_domain_name = 'mock_project_domain_name'
+        mock_user_domain_name = 'mock_user_domain_name'
+        mock_auth_token = 'mock_token123'
+        mock_remote_address = 'mock_addr'
+        mock_service_catalog = 'mock_service_catalog'
+        mock_req_id = 'mock_req_id'
+
+        req_mock.remote_addr = mock_remote_address
+        mock_jsonutils_loads.return_value = mock_service_catalog
+        req_mock.environ[request_id.ENV_REQUEST_ID] = mock_req_id
+
+        req_mock.headers = {
+            'X_USER': mock_user,
+            'X_USER_ID': mock_id,
+            'X_ROLE': '1,2,3',
+            'X_TENANT_ID': mock_project_id,
+            'X_TENANT': 'mock_tenant',
+            'X_TENANT_NAME': mock_project_name,
+            'X-Project-Domain-Name': mock_project_domain_name,
+            'X-User-Domain-Name': mock_user_domain_name,
+            'X_AUTH_TOKEN': mock_auth_token,
+            'X_SERVICE_CATALOG': 'mock_catalog',
+        }
+
+        mock_roles = [
+            r.strip() for r in req_mock.headers.get('X_ROLE', '').split(',')
+        ]
+
+        result = CoriolisKeystoneContext(wsgi.Middleware)
```

Comment:
You can `__call__` by `result = CoriolisKeystoneContext(wsgi.Middleware)(req_mock)`